mypy/fastparse.py

```diff
@@ -956,8 +956,9 @@ def visit_Subscript(self, n: ast35.Subscript) -> Type:
 
         value = self.visit(n.value)
 
-        assert isinstance(value, UnboundType)
-        assert not value.args
+        if not isinstance(value, UnboundType) or value.args:
```

Comment:
The logic here is a bit obfuscated, esp. because "if not X or Y" reads confusingly. I'd recommend changing things around so it's more like `if isinstance(value, UnboundType) and not value.args: ...good stuff... else: ...error...`